src/libraries/System.Text.Json/tests/Common/RequiredKeywordTests.cs

```diff
@@ -664,6 +664,27 @@ public class ClassWithCustomRequiredPropertyName
             public required int PropertyWithInitOnlySetter { get; init; }
         }
 
+        [Fact]
+        public async Task DerivedClassWithRequiredProperty()
+        {
+            var value = new DerivedClassWithRequiredInitOnlyProperty { MyInt = 42 };
+            string json = await Serializer.SerializeWrapper(value);
+            Assert.Equal("""{"MyInt":42}""", json);
+
+            value = await Serializer.DeserializeWrapper<DerivedClassWithRequiredInitOnlyProperty>(json);
+            Assert.Equal(42, value.MyInt);
+        }
+
+        public class BaseClassWithInitOnlyProperty
+        {
+            public int MyInt { get; init; }
+        }
+
+        public class DerivedClassWithRequiredInitOnlyProperty : BaseClassWithInitOnlyProperty
+        {
+            public new required int MyInt { get; init; }
```

Comment:
Would it be possible to also add test cases for `required` only or `init` only properties? They don't have to be on a separate test, you could just add more properties in that particular class.

Reply:
Test extended.